lib/google/gax/api_call.rb

```diff
@@ -158,7 +158,8 @@ def call request, options: nil, format_response: nil, operation_callback: nil, s
             retry if options.retry_policy.call error
           end
 
-          error = Google::Gax.from_error(error).new "RPC failed" if error.is_a? GRPC::BadStatus
+          # Wrap error in GaxError if GRPC::BadStatus
+          error = Google::Gax::GaxError.wrap error
```

Comment:
nit: this reads less clear than the original to me. I prefer the check here and perhaps a raise if the error is inappropriate. (I'm pointing this out due to my other comment about keeping this public, otherwise it doesn't really matter.)

Reply:
This is the only place this method is used. Perhaps we should move the implementation from `Google::Gax::GaxError` to a private method on `Google::Gax::ApiCall`?

Reply:
Your call. That sounds better to me though.

Reply:
Reverted.

Reply:
Added `wrap_error` as a private method.